torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -173,10 +173,9 @@ bool validateKernelArgTensor(
   }
   // Check the rank of the tensors.
   size_t arg_dim = arg.dim();
-  size_t param_dim = static_cast<const TensorView*>(param)
-                         ->getRootDomain()
-                         ->noReductions()
-                         ->nDims();
+  size_t param_dim = TensorDomain::noReductions(
+                         static_cast<const TensorView*>(param)->getRootDomain())
```

Comment:
param->as<TensorView>()->....

Reply:
I think we can go through in another PR cleaning up all my static_casts if that's desired.

Reply:
Made a note on https://github.com/csarofeen/pytorch/issues/34 to run through and clean static_casts up.